yaml/template.go

```diff
@@ -19,6 +19,7 @@ type (
 	Template struct {
 		Name   string `yaml:"name,omitempty"   json:"name,omitempty"  jsonschema:"required,minLength=1,description=Unique identifier for the template.\nReference: https://go-vela.github.io/docs/concepts/pipeline/templates/"`
 		Source string `yaml:"source,omitempty" json:"source,omitempty" jsonschema:"required,minLength=1,description=Path to template in remote system.\nReference: https://go-vela.github.io/docs/concepts/pipeline/templates/source/"`
+		Format string `yaml:"format,omitempty"   json:"format,omitempty" jsonschema:"minLength=1,description=language used within the template file \nReference: https://go-vela.github.io/docs/concepts/pipeline/templates/format/,example=go,starlark"`
```

Comment:
i think we might want an enum definition here if there is a finite two options for this right now. see https://github.com/go-vela/types/blob/v0.6.0/yaml/step.go#L32 for an example.

Reply:
Right now, we have it set up as mainly two options but one has different permutations:

`format: starlark` will ensure the template gets expanded with that code path.

Go however has several options since we wanted to ensure backward compatibility for all existing templates. So, the following are valid to use a Go template:

```
format: golang
format: go
format: ""
```